java/code/src/com/suse/manager/reactor/messaging/RegisterMinionEventMessageAction.java

```diff
@@ -533,19 +539,26 @@ else if (!minion.getOrg().equals(org)) {
             }
 
             // Saltboot treatment - prepare and apply saltboot
-            if (grains.getOptionalAsBoolean("saltboot_initrd").orElse(false)) {
+            if (saltbootInitrd) {
                 LOG.info("\"saltboot_initrd\" grain set to true: Preparing & applying saltboot for minion {}",
                         minionId);
-                prepareRetailMinionForSaltboot(minion, org, grains);
-                applySaltboot(minion);
+                try {
+                    prepareRetailMinionForSaltboot(minion, org, grains);
+                    applySaltboot(minion, false);
+                }
+                catch (Throwable t) {
+                    LOG.error("Error preparing saltboot configuration for minion id: {}", minionId, t);
+                    // do not throw, leave the minion as registered in initrd
```

Comment:
This is just log error, but at this moment we already have server entry in database and by not throwing this entry will not be removed. Can we add failed event to the event log as well?

Reply:
Ah, nevermind. They are added in the methods themselves.